src/lib/swapper/swappers/ThorchainSwapper/evm/makeTradeTx.ts

```diff
@@ -67,14 +68,25 @@ export const makeTradeTx = async ({
 
     const { data, router } = thorTxInfo
 
+    const { feesWithGasLimit } = await getFeesFromContractData({
+      accountNumber,
+      adapter,
+      to: router,
+      data,
+      wallet,
+    })
+
     return Ok(
       await adapter.buildCustomTx({
         wallet,
         accountNumber,
         to: router,
         value: isErc20Trade ? '0' : sellAmountCryptoBaseUnit,
         data,
-        ...(await getFeesFromFeeData({ wallet, feeData: feeData.chainSpecific })),
+        ...feesWithGasLimit,
+        // TODO: implement dynamic gas limit
+        // https://github.com/shapeshift/web/issues/4512
+        gasLimit: THOR_EVM_GAS_LIMIT,
```

Comment:
don't we already have a real `gasLimit` being returned on `feesWithGasLimit`?

Reply:
https://github.com/shapeshift/web/pull/4492/commits/d99fd65ebfd9686795e9dfc5612c7e2ae48b3922

Reply:
we cant remove this, the gas limit returned in `feesWithGasLimit` is incorrect for thorchain

Reply:
> https://etherscan.io/tx/0x82b4d7a2abfed1d7c76a10bc181e6c625c292700a4e152cc2da9a48390d405dc

by passing through the correct value, we estimate correctly now

Reply:
have a read of the docs linked in https://github.com/shapeshift/web/issues/4512 for more details, the gas limit reqs here are thorchain specific :)

Reply:
fixed 9df283dbb49d24b0bbcacc11940f11960e94a6f7

Reply:
yeah I read it, ideally we use our fees and can estimate gas limit via the node instead of using A) their gas estimation of `fast`, and B) the dumbed down way to calculate gasLimit that doesn't take into account the full evm call stack. https://growingdata.com.au/how-to-calculate-gas-fees-on-ethereum/